hw/top_earlgrey/data/chip_testplan.hjson

```diff
@@ -432,45 +433,55 @@
     {
       name: chip_rv_dm_cpu_debug_mem
       desc: '''Verify access to the debug mem from the CPU.
+
+            - CPU programs RW type CSRs (such as progbuf0 and progbuf1) with random value.
+            - Read back to check their values for correctness.
             '''
       milestone: V2
       tests: []
     }
     {
       name: chip_rv_dm_jtag_debug_mem
       desc: '''Verify access to the debug mem from the external JTAG interface.
+
+            - Load debug ROM and read back through jtag to check the values for correctness.
             '''
       milestone: V2
       tests: []
     }
     {
       name: chip_rv_dm_cpu_debug_req
       desc: '''Verify debug request to Ibex while it is actively executing.
+
+            - While Ibex is executing, trigger a debug request through programming
+              `dmcontrol.haltreq`.
+            - Ibex steps into interactive debug to store a value in SRAM.
+            - After Ibex jumps back to the program execution, check the stored value in the SRAM.
             '''
       milestone: V2
       tests: []
     }
     {
       name: chip_rv_dm_ndm_reset_req
-      desc: '''Verify non-debug reset req initiated from RV_DM when the chip is awake.
+      desc: '''Verify non-debug reset request initiated from RV_DM when the chip is awake.
 
-            Read CSRs / mem within all IPs in the chip to ensure that they are reset to the original
-            values. Read CSRs / mem in the debug domain to ensure that the values survive the reset.
+            - Program some CSRs / mem that are under life cycle reset tree and system reset tree.
+            - Configure RV_DM to send NDM reset request to reset sytem reset tree.
+            - Read the programmed CSRs / mem to ensure that everything under system reset tree is
+              reseted to the original values, while values under life cycle reset will be
+              preserved.
+            - Read CSRs / mem in the debug domain to ensure that the values survive the reset.
+            X-ref'ed with the rstmgr tests
```

Comment:
Please add a "TODO" here -- or do these tests already exist?

Reply:
I don't find that test in rstmgr. Added TODO.

Reply:
I've just been using the debug module with a simple Ibex based system and think I've found a bug with the non-debug reset functionality. If we're in the halted state (where we're in debug mode in Ibex and anyhalted/allhalted will be set in the DMI dmstatus debug register) and issue an non-debug reset the debug module still believes Ibex is halted.

Still looking at a fix but need to make sure this case is covered here. I think you want to test the non-debug reset both when Ibex is and isn't halted (from the DM point of view). You want to check the debug module doesn't leave anyhalted/allhalted set after the reset.

Reply:
@GregAC I don't mind either way, but it seems to me that this test should go into the block-level test, not the chip-level test. That said, I don't think we even have a IP level testplan yet for rv_dm.

Reply:
Lets add a testpoint for the scanario @GregAC pointed out at the chip level as well. 

As for the `Xref ` line, lets just remove it to avoid confusion.